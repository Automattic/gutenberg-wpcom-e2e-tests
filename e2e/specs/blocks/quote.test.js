/**
 * Internal dependencies
 */
import {
	clickBlockAppender,
	getEditedPostContent,
	newPost,
	pressTimes,
	convertBlock,
	insertBlock,
} from '../../support/utils';

describe( 'Quote', () => {
	beforeEach( async () => {
		await newPost();
	} );

	it( 'can be created by using > at the start of a paragraph block', async () => {
		// Create a block with some text that will trigger a list creation.
		await clickBlockAppender();
		await page.keyboard.type( '> A quote' );

		// Create a second list item.
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'Another paragraph' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by typing > in front of text of a paragraph block', async () => {
		// Create a list with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await pressTimes( 'ArrowLeft', 4 );
		await page.keyboard.type( '> ' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by typing "/quote"', async () => {
		// Create a list with the slash block shortcut.
		await clickBlockAppender();
		await page.keyboard.type( '/quote' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'Iâ€™m a quote' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by converting a paragraph', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'test' );
		await convertBlock( 'Quote' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be created by converting multiple paragraphs', async () => {
		await clickBlockAppender();
		await page.keyboard.type( 'one' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'two' );
		await page.keyboard.down( 'Shift' );
		await page.click( '[data-type="core/paragraph"]' );
		await page.keyboard.up( 'Shift' );
		await convertBlock( 'Quote' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	describe( 'can be converted to paragraphs', async () => {
		it( 'and renders one paragraph block per <p> within quote', async () => {
			await insertBlock( 'Quote' );
			await page.keyboard.type( 'one' );
			await page.keyboard.press( 'Enter' );
			await page.keyboard.type( 'two' );
			await convertBlock( 'Paragraph' );

			expect( await getEditedPostContent() ).toMatchSnapshot();
		} );

		it( 'and renders a paragraph for the cite, if it exists', async () => {
			await insertBlock( 'Quote' );
			await page.keyboard.type( 'one' );
			await page.keyboard.press( 'Enter' );
			await page.keyboard.type( 'two' );
			await page.keyboard.press( 'Tab' );
			await page.keyboard.type( 'cite' );
			await convertBlock( 'Paragraph' );

			expect( await getEditedPostContent() ).toMatchSnapshot();
		} );

		it( 'and renders only one paragraph for the cite, if the quote is void', async () => {
			await insertBlock( 'Quote' );
			await page.keyboard.press( 'Tab' );
			await page.keyboard.type( 'cite' );
			await convertBlock( 'Paragraph' );

			expect( await getEditedPostContent() ).toMatchSnapshot();
		} );

		it( 'and renders a void paragraph if both the cite and quote are void', async () => {
			await insertBlock( 'Quote' );
			await convertBlock( 'Paragraph' );

			expect( await getEditedPostContent() ).toMatchSnapshot();
		} );
	} );

	it( 'can be created by converting a heading', async () => {
		await insertBlock( 'Heading' );
		await page.keyboard.type( 'test' );
		await convertBlock( 'Quote' );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be converted to headings', async () => {
		await insertBlock( 'Quote' );
		await page.keyboard.type( 'one' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'two' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'cite' );
		await convertBlock( 'Heading' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
		await page.click( '[data-type="core/quote"]' );
		await convertBlock( 'Heading' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
		await page.click( '[data-type="core/quote"]' );
		await convertBlock( 'Heading' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be converted to a pullquote', async () => {
		await insertBlock( 'Quote' );
		await page.keyboard.type( 'one' );
		await page.keyboard.press( 'Enter' );
		await page.keyboard.type( 'two' );
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'cite' );
		await convertBlock( 'Pullquote' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'can be merged into from a paragraph', async () => {
		await insertBlock( 'Quote' );
		await insertBlock( 'Paragraph' );
		await page.keyboard.type( 'test' );
		await pressTimes( 'ArrowLeft', 'test'.length );
		await page.keyboard.press( 'Backspace' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
