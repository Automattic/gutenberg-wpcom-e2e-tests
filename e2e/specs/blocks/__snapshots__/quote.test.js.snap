// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote can be converted to a pullquote 1`] = `
"<!-- wp:pullquote -->
<figure class=\\"wp-block-pullquote\\"><blockquote><p>one</p><p>two</p><cite>cite</cite></blockquote></figure>
<!-- /wp:pullquote -->"
`;

exports[`Quote can be converted to headings 1`] = `
"<!-- wp:heading -->
<h2>one</h2>
<!-- /wp:heading -->

<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>two</p><cite>cite</cite></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be converted to headings 2`] = `
"<!-- wp:heading -->
<h2>one</h2>
<!-- /wp:heading -->

<!-- wp:heading -->
<h2>two</h2>
<!-- /wp:heading -->

<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p></p><cite>cite</cite></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be converted to headings 3`] = `
"<!-- wp:heading -->
<h2>one</h2>
<!-- /wp:heading -->

<!-- wp:heading -->
<h2>two</h2>
<!-- /wp:heading -->

<!-- wp:heading -->
<h2>cite</h2>
<!-- /wp:heading -->"
`;

exports[`Quote can be converted to paragraphs and renders a paragraph for the cite, if it exists 1`] = `
"<!-- wp:paragraph -->
<p>one</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>two</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>cite</p>
<!-- /wp:paragraph -->"
`;

exports[`Quote can be converted to paragraphs and renders a void paragraph if both the cite and quote are void 1`] = `
"<!-- wp:paragraph -->
<p></p>
<!-- /wp:paragraph -->"
`;

exports[`Quote can be converted to paragraphs and renders one paragraph block per <p> within quote 1`] = `
"<!-- wp:paragraph -->
<p>one</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>two</p>
<!-- /wp:paragraph -->"
`;

exports[`Quote can be converted to paragraphs and renders only one paragraph for the cite, if the quote is void 1`] = `
"<!-- wp:paragraph -->
<p>cite</p>
<!-- /wp:paragraph -->"
`;

exports[`Quote can be created by converting a heading 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>test</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be created by converting a paragraph 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>test</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be created by converting multiple paragraphs 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>one</p><p>two</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be created by typing "/quote" 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>Iâ€™m a quote</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be created by typing > in front of text of a paragraph block 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>test</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be created by using > at the start of a paragraph block 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>A quote</p><p>Another paragraph</p></blockquote>
<!-- /wp:quote -->"
`;

exports[`Quote can be merged into from a paragraph 1`] = `
"<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>test</p></blockquote>
<!-- /wp:quote -->"
`;
